packages/common/src/store/pages/chat/middleware.ts

```diff
@@ -0,0 +1,73 @@
+import { ChatEvents, type sdk } from '@audius/sdk'
+import { Middleware } from 'redux'
+
+import { actions as chatActions } from './slice'
+
+const {
+  connect,
+  disconnect,
+  addMessage,
+  incrementUnreadCount,
+  setMessageReactionSucceeded
+} = chatActions
+
+export const chatMiddleware =
+  (audiusSdk: () => Promise<ReturnType<typeof sdk>>): Middleware =>
+  (store) => {
+    let messageListener: ChatEvents['message'] | null = null
+    let reactionListener: ChatEvents['reaction'] | null = null
+    let openListener: ChatEvents['open'] | null = null
+    let closeListener: ChatEvents['close'] | null = null
+
+    return (next) => (action) => {
+      if (connect.match(action)) {
+        console.log('Chat middleware attaching')
+        audiusSdk().then((sdk) => {
```

Comment:
nit: any reason using promises vs async await?

Reply:
I suppose I could wrap the contents in a callable async fn. I don't think the middleware should be async, though.